provisioning/baremetal_webhook.go

```diff
@@ -31,15 +31,14 @@ func EnsureBaremetalOperatorWebhook(info *ProvisioningInfo) (bool, error) {
 	}
 
 	webhookService := newBaremetalOperatorWebhookService(info.Namespace)
-	_, _, err := resourceapply.ApplyService(info.Client.CoreV1(), info.EventRecorder, webhookService)
+	_, _, err := resourceapply.ApplyService(context.Background(), info.Client.CoreV1(), info.EventRecorder, webhookService)
 	if err != nil {
 		err = errors.Wrap(err, "unable to create validatingwebhook service")
 		return false, err
 	}
 
 	vw := newBaremetalOperatorWebhook(info.Namespace)
-	expectedGeneration := resourcemerge.ExpectedValidatingWebhooksConfiguration(validatingWebhookConfigurationName, info.ProvConfig.Status.Generations)
-	validatingWebhook, updated, err := resourceapply.ApplyValidatingWebhookConfiguration(info.Client.AdmissionregistrationV1(), info.EventRecorder, vw, expectedGeneration)
+	validatingWebhook, updated, err := resourceapply.ApplyValidatingWebhookConfigurationImproved(context.Background(), info.Client.AdmissionregistrationV1(), info.EventRecorder, vw, resourceapply.NewResourceCache())
```

Comment:
I wonder if this change is somehow breaking. I don't really understand generations well enough to judge.

Reply:
From what I could tell looking into the library-go commit history, the generation parameter was only really intended to workaround some issue with deployments and apparently didn't apply to webhooks.

I should be able to find the commit in a bit.

Reply:
That said, I'm less sure about the cache parameter.

I'll look into that a bit more and see if we should be doing something more clever.